### PR TITLE
RFC: Sub Team Maintainer (formerly author) Creates Repo Issues

### PR DESCRIPTION
**Pull request by dfreilich**
[Readable](https://github.com/dfreilich/rfcs/blob/make-issues/text/0000-create-repo-issues.md)

Signed-off-by: David Freilich <dfreilich@vmware.com>

### CODE REVIEW

File: text/0000-create-repo-issues.md
Comment (hone) [inline text/0000-create-repo-issues.md:90]: This detail is great, but as a FYI the RFC process is not part of the spec so would not need a spec change.

**Comment by hone** [comment]
I think this makes a lot of sense if the RFC comes from a contributor/core team member, but I've noticed in the prior art that none of the other RFC processes follow this practice. I wonder if it this instead should fall upon the subteam to manage/create issues when they're finalized maybe in the subteam/maintainers meeting once a week to go over finalized RFCs. This requires a lot more onus on the RFC author especially if they are a non-contributor.

**Comment by dfreilich** [comment]
@hone I definitely hear that. A suggestion that came up in conversation (I believe from @natalieparellano) was to add a section like
```
# Suggested Issues
```
to the RFC template, where the author of the PR would write what repos/issues they would create, which could then get approval from the reviewers, as well as additional suggestions. 

To lower the barrier of entry, I would add that while we welcome well-fleshed out issues, even placeholder ones that link to the RFC are a better start than no issues.

File: text/0000-create-repo-issues.md
Comment (ekcasey) [inline text/0000-create-repo-issues.md:30]: I am not sure that I agree that "the person who proposed the change is the best person to create the issue". I fear that adding requirements like this makes the RFC process even more unapproachable for new contributors. I believe the best person to create the issue is a maintainer of the relevant repo. 

As an implementation maintainer I currently try to create spec and lifecycle issues as needed when RFCs merge (examples: https://github.com/buildpacks/spec/issues/106, https://github.com/buildpacks/lifecycle/issues/354).

This makes sense to me b/c I have the lifecycle knowledge and permissions required to subdivide, label, organize in milestones etc. Example: RFC-0053 affected two APIs therefore needs two separate spec issues ( https://github.com/buildpacks/spec/issues/134, https://github.com/buildpacks/spec/issues/133) I don't think think every RFC author should have to think about that level of detail (although stephen wrote this one and he certainly could have ;p).

On the other hand I _do_ think every subteam maintainer should be thinking about what each RFC means for their subteam as RFCs go through and should be planning/organizing accordingly.

Comment (hone) [reply text/0000-create-repo-issues.md:30]: This mirrors my comment above. I'm very much in favor of issues being created from RFCs, but I think this should be handled by the subteams (not just the maintainers). Would you be open to that change @dfreilich?

Comment (dfreilich) [reply text/0000-create-repo-issues.md:30]: I'd definitely be open to that, that makes sense. One suggestion - do people think it makes sense to add additional label(s), like `needs-issues-impl`, `needs-issues-platform`, etc? In that context, when an RFC is put in FCP, the appropriate labels could be added, and they'd need to be removed for an RFC to be merged in. Or would that be over-engineering the solution...

**Review by nebhale — APPROVED** [review]

**Review by jkutner — APPROVED** [review]

File: text/0000-create-repo-issues.md
Comment (ekcasey) [inline text/0000-create-repo-issues.md:56]: As part of the merge process, should the core team member add references to the created issues in the `CNB Issue` field in the RFC meta section?

Maybe I am misunderstanding the original intention of this field (cc @hone)? In practice, we seem to leave it blank most of the time, but it could be powerful navigational tool.

Example: https://github.com/buildpacks/rfcs/blob/main/text/0044-pack-publish-buildpack.md#meta

File: text/0000-create-repo-issues.md
Comment (ekcasey) [inline text/0000-create-repo-issues.md:40]: Given that some sub-teams have multiple maintainers, there may still be some potential for ambiguity around who will handle a particular RFC. I propose that when we move RFCs into FCP we should assign a sub-team maintainer to the PR and they will be responsible for creating the issues.

Comment (hone) [reply text/0000-create-repo-issues.md:40]: @ekcasey my thought is that this would happen during the weekly subteam meetings, but I don't feel strongly about it.

File: text/0000-create-repo-issues.md
Comment (hone) [reply text/0000-create-repo-issues.md:56]: yeah, that's the intention so we can reference the RFC discussion after it's been merged.

Comment (dfreilich) [reply text/0000-create-repo-issues.md:56]: Great point, @ekcasey. I'll add that in.

File: text/0000-create-repo-issues.md
Comment (dfreilich) [reply text/0000-create-repo-issues.md:40]: @ekcasey I don't know whether we need that level of process at this point. I agree with @hone that I had felt it could be done during the subteam meetings, as one of the repeating action items. Should I clarify that?

Comment (jromero) [reply text/0000-create-repo-issues.md]: There are some RFCs that require multiple sub-teams to create issues from a single RFC. For the sake of minimizing coordination could we maybe change this to be a label per subteam.

For example:

* `issues-created/platform`
* `issues-created/implementation`

If no issues need to be created for a given subteam they can add that label for brevity sake. We can further automate this process by checking that these labels are present.

Comment (dfreilich) [reply text/0000-create-repo-issues.md]: I incorporated @jromero's feedback. LMK what you think.

**Review by ekcasey — APPROVED** [review]

**Comment by dfreilich** [comment]
@nebhale and @jkutner I made some changes since you last checked it over. Just want to make sure you're still ok with the contents.

**Review by jkutner — APPROVED** [review]

**Review by nebhale — APPROVED** [review]

**Comment by hone** [comment]
Once this RFC is merged in that means someone from the core team will need to create the issue on this repo. :)

**Review by hone — APPROVED** [review]

**Comment by ekcasey** [comment]
Final Comment Period with merge disposition, closing on 30 September, 2020.

File: text/0000-create-repo-issues.md
Comment (jromero) [inline text/0000-create-repo-issues.md:40]:
```suggestion
When a PR has been accepted (after FCP has concluded), the sub-team maintainers make issues (tagging the RFC PR) in the relevant buildpack repos, and labels the PR `issues-created/<sub-team>` for their respective sub-team. Once all sub-team `issues-created/<sub-team>` labels are set, the RFC PR may be merged in, with references to the CNB issues added to the RFC [metadata](https://github.com/buildpacks/rfcs/blob/main/text/0044-pack-publish-buildpack.md#meta).
```

Comment (dfreilich) [reply text/0000-create-repo-issues.md:40]: Signed-off-by: David Freilich <dfreilich@vmware.com>

**Pull request merged**
